core-plugins/src/main/java/co/cask/hydrator/plugin/common/KinesisOutputFormat.java

```diff
@@ -0,0 +1,160 @@
+    return new KinesisRecordWriter(kinesisClient, context);
+  }
+
+  /**
+   * Check for validity of the output-specification for the job and initialize {@link AmazonKinesisClient}
+   * <p>
+   * <p>This is to validate the output specification for the job when it is
+   * a job is submitted.</p>
+   *
+   * @param context information about the job
+   * @throws IOException when output should not be attempted
+   */
+  @Override
+  public void checkOutputSpecs(JobContext context) throws IOException, InterruptedException {
+    Configuration conf = context.getConfiguration();
+    BasicAWSCredentials awsCred = new BasicAWSCredentials(conf.get(Properties.KinesisRealtimeSink.ACCESS_ID),
+                                                          conf.get(Properties.KinesisRealtimeSink.ACCESS_KEY));
+
+    if (kinesisClient == null) {
+      LOG.info("Creating amazon kinesis client for stream {}", Properties.KinesisRealtimeSink.NAME);
+      kinesisClient = new AmazonKinesisClient(awsCred);
+      try {
+        KinesisUtil.createAndWaitForStream(kinesisClient, conf.get(Properties.KinesisRealtimeSink.NAME),
+                                           Integer.valueOf(conf.get(Properties.KinesisRealtimeSink.SHARD_COUNT)));
+      } catch (AmazonServiceException e) {
+        if (e.getStatusCode() == 400 && e.getErrorCode().equalsIgnoreCase("UnrecognizedClientException")) {
+          throw new IllegalArgumentException("AWS credentials are not correct", e);
+        } else {
+          throw e;
+        }
+      }
+    }
+  }
+
+  /**
+   * Get the output committer for this output format. This is responsible
+   * for ensuring the output is committed correctly.
+   *
+   * @param context the task context
+   * @return an output committer
+   * @throws IOException
+   * @throws InterruptedException
+   */
+  @Override
+  public OutputCommitter getOutputCommitter(TaskAttemptContext context) throws IOException, InterruptedException {
+    return new NoOpOutputCommitter();
+  }
+
+  private class KinesisRecordWriter extends RecordWriter<NullWritable, Text> {
+    private final AmazonKinesisClient kinesisClient;
+    private final TaskAttemptContext context;
+    private final boolean distribute;
+
+    KinesisRecordWriter(AmazonKinesisClient kinesisClient, TaskAttemptContext context) {
+      this.kinesisClient = kinesisClient;
+      this.context = context;
+      distribute = "true".equals(context.getConfiguration().get(Properties.KinesisRealtimeSink.DISTRIBUTE));
+    }
+
+    /**
+     * Writes a key/value pair.
+     *
+     * @param key   the key to write.
+     * @param value the value to write.
+     * @throws IOException
+     */
+    @Override
+    public void write(NullWritable key, Text value) throws IOException, InterruptedException {
+      PutRecordRequest putRecordRequest = new PutRecordRequest();
+      putRecordRequest.setStreamName(context.getConfiguration().get(Properties.KinesisRealtimeSink.NAME));
```

Comment:
put the stream name in a class field, since it wont change each call. (no need to get the property each time).